oauthproxy.go

```diff
@@ -146,6 +153,23 @@ func NewFileServer(path string, filesystemPath string) (proxy http.Handler) {
 	return http.StripPrefix(path, http.FileServer(http.Dir(filesystemPath)))
 }
 
+func NewWebSocketOrRestReverseProxy(u *url.URL, opts *Options, auth hmacauth.HmacAuth) (restProxy http.Handler) {
+	u.Path = ""
+	proxy, err := NewReverseProxy(u, opts.UpstreamFlush, opts.UpstreamCAs)
+	if err != nil {
+		log.Fatal("Failed to initialize Reverse Proxy: ", err)
+	}
+	if !opts.PassHostHeader {
+		setProxyUpstreamHostHeader(proxy, u)
+	} else {
+		setProxyDirector(proxy)
+	}
+
+	wsURL := &url.URL{Scheme: "ws://", Host: u.Host}
```

Comment:
We should support wss also but that support can follow this PR.

Reply:
Added some initial support for it.  Should use ws or wss based on if it's an http or https upstream.